src/main/java/net/javadiscord/javabot/systems/user_commands/leaderboard/QOTWLeaderboardSubcommand.java

```diff
@@ -0,0 +1,205 @@
+		int x = left ? MARGIN * 5 : WIDTH - (MARGIN * 5) - card.getWidth();
+		g2d.drawImage(ImageGenerationUtils.getImageFromUrl(member.getUser().getEffectiveAvatarUrl() + "?size=4096"), x + 185, y + 43, 200, 200, null);
+		String displayName = member.getUser().getAsTag();
+		// draw card
+		g2d.drawImage(card, x, y, null);
+		g2d.setColor(PRIMARY_COLOR);
+		g2d.setFont(ImageGenerationUtils.getResourceFont("assets/fonts/Uni-Sans-Heavy.ttf", 65).orElseThrow());
+
+		int stringWidth = g2d.getFontMetrics().stringWidth(displayName);
+		while (stringWidth > 750) {
+			var currentFont = g2d.getFont();
+			var newFont = currentFont.deriveFont(currentFont.getSize() - 1F);
+			g2d.setFont(newFont);
+			stringWidth = g2d.getFontMetrics().stringWidth(displayName);
+		}
+		g2d.drawString(displayName, x + 430, y + 130);
+		g2d.setColor(SECONDARY_COLOR);
+		g2d.setFont(ImageGenerationUtils.getResourceFont("assets/fonts/Uni-Sans-Heavy.ttf", 72).orElseThrow());
+
+		var points = service.getPoints(member.getIdLong());
+		String text = points + (points > 1 ? " points" : " point");
+		String rank = "#" + service.getQOTWRank(member.getIdLong());
+		g2d.drawString(text, x + 430, y + 210);
+		int stringLength = (int) g2d.getFontMetrics().getStringBounds(rank, g2d).getWidth();
+		int start = 185 / 2 - stringLength / 2;
+		g2d.drawString(rank, x + start, y + 173);
+	}
+
+	/**
+	 * Draws and constructs the leaderboard image.
+	 *
+	 * @param guild   The current guild.
+	 * @param service The {@link QOTWPointsService}.
+	 * @return The finished image as a {@link ByteArrayInputStream}.
+	 * @throws IOException If an error occurs.
+	 */
+	private ByteArrayOutputStream generateLeaderboard(Guild guild, QOTWPointsService service) throws IOException {
+		BufferedImage logo = ImageGenerationUtils.getResourceImage("assets/images/QuestionOfTheWeekHeader.png");
+		BufferedImage card = ImageGenerationUtils.getResourceImage("assets/images/LeaderboardUserCard.png");
+
+		List<Member> topMembers = service.getTopMembers(DISPLAY_COUNT, guild);
+		int height = (logo.getHeight() + MARGIN * 3) +
+				(ImageGenerationUtils.getResourceImage("assets/images/LeaderboardUserCard.png").getHeight() + MARGIN) * (Math.min(DISPLAY_COUNT, topMembers.size()) / 2) + MARGIN;
+		BufferedImage image = new BufferedImage(WIDTH, height, BufferedImage.TYPE_INT_RGB);
+		Graphics2D g2d = image.createGraphics();
+
+		g2d.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_LCD_HRGB);
+		g2d.setRenderingHint(RenderingHints.KEY_FRACTIONALMETRICS, RenderingHints.VALUE_FRACTIONALMETRICS_ON);
+		g2d.setPaint(BACKGROUND_COLOR);
+		g2d.fillRect(0, 0, WIDTH, height);
+		g2d.drawImage(logo, WIDTH / 2 - logo.getWidth() / 2, MARGIN, null);
+
+		boolean left = true;
+		int y = logo.getHeight() + 3 * MARGIN;
+		for (Member m : topMembers) {
+			drawUserCard(g2d, m, service, y, left);
+			left = !left;
+			if (left) y = y + card.getHeight() + MARGIN;
+		}
+		g2d.dispose();
```

Comment:
You might want to use a `try-with-resources` block for this.

Reply:
What, where? Graphics2D is not AutoClosable